### ISSUE TITLE
Snapshot folder is gone

### ISSUE THREAD

**Issue by Luc3as** [sabeechen/hassio-google-drive-backup#300]
Copy the info from the previous page here
###### Description:
```
The strange behaviour, after syncing one of old local snapshots to google drive, I got error message that the folder is gone, however, I can access it with the same ID, I can see uploaded snapshot, and even when I go to the addon UI and enter the same folder UI (folder selector is not working from some reason) the sync starts again without any problems. it looks like the id is not stored somewhere it should. 
```

 Addon version: `0.102.0`
 Home Assistant Version: `0.115.6`
 Supervisor Version: `2020.10.0`
 Supervisor Channel: `stable`
 Hassos Version: `None`
 Docker Version: `19.03.13`
 Architecture: `amd64`
 Machine: `qemux86-64`
 Date: `2020-11-02T13:46:43.967810+00:00`
 Timezone: `Europe/Bratislava`
 Failure Time: `2020-11-02T13:31:00.176752+00:00`
 Last Good Sync: `2020-11-02T13:46:21.009888+00:00`
 ###### Exception:
 ```
 
  addon/backup/model/coordinator.py:148 (_sync)
  addon/backup/model/model.py:136 (sync)
  addon/backup/model/model.py:221 (_syncSnapshots)
  addon/backup/drive/drivesource.py:84 (get)
  addon/backup/drive/drivesource.py:189 (getFolderId)
  addon/backup/drive/folderfinder.py:58 (get)
  addon/backup/drive/folderfinder.py:110 (_readFolderId)
backup.exceptions.exceptions.BackupFolderInaccessible: 1rlIVXUREDuUl8_Btrt6oj5BRpXSNxFac

 ```
 Snapshots:
 ```
 {
    "HomeAssistant": {
        "snapshots": 20,
        "retained": 0,
        "deletable": 20,
        "name": "HomeAssistant",
        "latest": "2020-11-02T11:10:39Z",
        "size": "60.3 GB"
    },
    "GoogleDrive": {
        "snapshots": 5,
        "retained": 0,
        "deletable": 5,
        "name": "GoogleDrive",
        "latest": "2020-11-02T11:10:39Z",
        "size": "15.1 GB"
    }
}
 ```
 ###### Config:
 ```
 {
    "Setting.MAX_SNAPSHOTS_IN_HASSIO": 20,
    "Setting.MAX_SNAPSHOTS_IN_GOOGLE_DRIVE": 20,
    "Setting.DAYS_BETWEEN_SNAPSHOTS": 1.0,
    "Setting.SNAPSHOT_TIME_OF_DAY": "02:00",
    "Setting.SNAPSHOT_PASSWORD": "REDACTED",
    "Setting.SPECIFY_SNAPSHOT_FOLDER": true,
    "Setting.EXCLUDE_ADDONS": "15ef4d2f_esphome-dev,a0d7b954_jupyterlablite,40817795_jupyterlablite,core_almond,core_ada",
    "Setting.USE_SSL": true,
    "Setting.CERTFILE": "fullchain.pem",
    "Setting.KEYFILE": "privkey.pem",
    "Setting.SEND_ERROR_REPORTS": true
}
 ```
 ###### Addon Logs:
 ```
 11-02 14:30:58 INFO [backup.drive.folderfinder] Provided snapshot folder 1rlIVXUREDuUl8_Btrt6oj5BRpXSNxFac is gone
11-02 14:30:58 ERROR [backup.model.coordinator] The choosen backup folder has become inaccessible.  Please visit the addon web UI to select a backup folder.
11-02 14:30:58 INFO [backup.model.coordinator] I'll try again in an hour
11-02 14:30:59 DEBUG [backup.model.syncer] Sync requested by GoogleDrive
11-02 14:30:59 INFO [backup.model.coordinator] Syncing Snapshots
11-02 14:30:59 DEBUG [backup.ha.harequests] Making Hassio request: http://hassio/snapshots
11-02 14:30:59 DEBUG [backup.drive.driverequests] Making Google Drive request: https://www.googleapis.com/drive/v3/files/1rlIVXUREDuUl8_Btrt6oj5BRpXSNxFac/?fields=id%2Cname%2CappProperties%2Csize%2Ctrashed%2CmimeType%2CmodifiedTime%2Ccapabilities%2Cparents&supportsAllDrives=true
11-02 14:31:00 INFO [backup.drive.folderfinder] Provided snapshot folder 1rlIVXUREDuUl8_Btrt6oj5BRpXSNxFac is gone
11-02 14:31:00 ERROR [backup.model.coordinator] The choosen backup folder has become inaccessible.  Please visit the addon web UI to select a backup folder.
11-02 14:31:00 INFO [backup.model.coordinator] I'll try again in an hour
11-02 14:46:20 DEBUG [backup.ha.harequests] Making Hassio request: http://hassio/addons/self/options
11-02 14:46:20 INFO [backup.drive.folderfinder] Saving snapshot folder: 1rlIVXUREDuUl8_Btrt6oj5BRpXSNxFac
11-02 14:46:21 DEBUG [backup.model.syncer] Sync requested by UI Server
11-02 14:46:21 INFO [backup.model.coordinator] Syncing Snapshots
11-02 14:46:21 DEBUG [backup.ha.harequests] Making Hassio request: http://hassio/snapshots
11-02 14:46:21 DEBUG [backup.drive.driverequests] Making Google Drive request: https://www.googleapis.com/drive/v3/files/?q=%271rlIVXUREDuUl8_Btrt6oj5BRpXSNxFac%27+in+parents&fields=nextPageToken%2Cfiles%28id%2Cname%2CappProperties%2Csize%2Ctrashed%2CmimeType%2CmodifiedTime%2Ccapabilities%2Cparents%29&pageSize=100&supportsAllDrives=true&includeItemsFromAllDrives=true&corpora=allDrives
11-02 14:46:21 INFO [backup.drive.drivesource] Uploading 'Partial Snapshot 2020-10-14 02:48:52' to Google Drive
11-02 14:46:21 DEBUG [backup.drive.driverequests] Starting a new upload session with Google Drive
11-02 14:46:21 DEBUG [backup.drive.driverequests] Making Google Drive request: https://www.googleapis.com/upload/drive/v3/files/?uploadType=resumable&supportsAllDrives=true
11-02 14:46:22 DEBUG [backup.drive.driverequests] Sending 262144 bytes to Google Drive
11-02 14:46:22 DEBUG [backup.drive.driverequests] Making Google Drive request: https://www.googleapis.com/upload/drive/v3/files/?uploadType=resumable&supportsAllDrives=true&upload_id=ABg5-UzxMBjhKgtLdFQZhuvfdLHeAYHxUGQOADtuJWRvEaQXDAXzIT6z39H5jVEr5A_XWHES438MevE-UgSGo4LufTbeXl-udQ
11-02 14:46:22 DEBUG [backup.drive.drivesource] Uploading Partial Snapshot 2020-10-14 02:48:52 0.01%
11-02 14:46:22 DEBUG [backup.drive.driverequests] Sending 8126464 bytes to Google Drive
11-02 14:46:22 DEBUG [backup.drive.driverequests] Making Google Drive request: https://www.googleapis.com/upload/drive/v3/files/?uploadType=resumable&supportsAllDrives=true&upload_id=ABg5-UzxMBjhKgtLdFQZhuvfdLHeAYHxUGQOADtuJWRvEaQXDAXzIT6z39H5jVEr5A_XWHES438MevE-UgSGo4LufTbeXl-udQ
11-02 14:46:32 DEBUG [backup.drive.drivesource] Uploading Partial Snapshot 2020-10-14 02:48:52 0.26%
11-02 14:46:32 DEBUG [backup.drive.driverequests] Sending 8126464 bytes to Google Drive
11-02 14:46:32 DEBUG [backup.drive.driverequests] Making Google Drive request: https://www.googleapis.com/upload/drive/v3/files/?uploadType=resumable&supportsAllDrives=true&upload_id=ABg5-UzxMBjhKgtLdFQZhuvfdLHeAYHxUGQOADtuJWRvEaQXDAXzIT6z39H5jVEr5A_XWHES438MevE-UgSGo4LufTbeXl-udQ
11-02 14:46:42 DEBUG [backup.drive.drivesource] Uploading Partial Snapshot 2020-10-14 02:48:52 0.51%
11-02 14:46:42 DEBUG [backup.drive.driverequests] Sending 8126464 bytes to Google Drive
11-02 14:46:42 DEBUG [backup.drive.driverequests] Making Google Drive request: https://www.googleapis.com/upload/drive/v3/files/?uploadType=resumable&supportsAllDrives=true&upload_id=ABg5-UzxMBjhKgtLdFQZhuvfdLHeAYHxUGQOADtuJWRvEaQXDAXzIT6z39H5jVEr5A_XWHES438MevE-UgSGo4LufTbeXl-udQ
 ```
 ###### Supervisor Logs:
 ```
 


[32m20-11-02 13:30:59 INFO (MainThread) [supervisor.api.security] /snapshots access from cebe7a76_hassio_google_drive_backup[0m
[32m20-11-02 13:31:41 INFO (MainThread) [supervisor.store.git] Update add-on https://github.com/esphome/hassio repository[0m
[32m20-11-02 13:31:41 INFO (MainThread) [supervisor.store.git] Update add-on https://github.com/hassio-addons/repository-edge repository[0m
[32m20-11-02 13:31:41 INFO (MainThread) [supervisor.store.git] Update add-on https://github.com/hassio-addons/repository repository[0m
[32m20-11-02 13:31:41 INFO (MainThread) [supervisor.store.git] Update add-on https://github.com/home-assistant/hassio-addons repository[0m
[32m20-11-02 13:31:41 INFO (MainThread) [supervisor.store.git] Update add-on https://github.com/sabeechen/hassio-google-drive-backup repository[0m
[32m20-11-02 13:31:41 INFO (MainThread) [supervisor.store.git] Update add-on https://github.com/hunterjm/hassio-addons repository[0m
[32m20-11-02 13:31:41 INFO (MainThread) [supervisor.store.git] Update add-on https://github.com/danielwelch/hassio-zigbee2mqtt repository[0m
[32m20-11-02 13:31:45 INFO (MainThread) [supervisor.store] Loading add-ons from store: 123 all - 0 new - 0 remove[0m

[32m20-11-02 13:34:01 INFO (MainThread) [supervisor.misc.tasks] Found new CoreDNS plugin version 2020.11.0, updating[0m
[32m20-11-02 13:34:01 INFO (SyncWorker_5) [supervisor.docker.interface] Updating image homeassistant/amd64-hassio-dns:2020.10.2 to homeassistant/amd64-hassio-dns:2020.11.0[0m
[32m20-11-02 13:34:01 INFO (SyncWorker_5) [supervisor.docker.interface] Downloading docker image homeassistant/amd64-hassio-dns with tag 2020.11.0.[0m
[32m20-11-02 13:34:03 INFO (SyncWorker_5) [supervisor.docker.interface] Stopping hassio_dns application[0m
[32m20-11-02 13:34:07 INFO (SyncWorker_5) [supervisor.docker.interface] Cleaning hassio_dns application[0m
[32m20-11-02 13:34:07 INFO (SyncWorker_1) [supervisor.docker.interface] Cleanup images: ['homeassistant/amd64-hassio-dns:2020.10.2'][0m
[32m20-11-02 13:34:07 INFO (MainThread) [supervisor.plugins.dns] Starting CoreDNS plugin[0m
[32m20-11-02 13:34:08 INFO (SyncWorker_0) [supervisor.docker.dns] Starting DNS homeassistant/amd64-hassio-dns with version 2020.11.0 - 172.30.32.3[0m


[32m20-11-02 13:45:23 INFO (MainThread) [supervisor.homeassistant.api] Updated Home Assistant API token[0m
[32m20-11-02 13:46:20 INFO (MainThread) [supervisor.homeassistant.secrets] Request secret snapshot_password[0m
[32m20-11-02 13:46:21 INFO (MainThread) [supervisor.api.security] /snapshots access from cebe7a76_hassio_google_drive_backup[0m
[32m20-11-02 13:46:21 INFO (MainThread) [supervisor.api.security] /snapshots/6dfbf986/download access from cebe7a76_hassio_google_drive_backup[0m
[32m20-11-02 13:46:21 INFO (MainThread) [supervisor.api.snapshots] Downloading snapshot 6dfbf986[0m
[32m20-11-02 13:46:43 INFO (MainThread) [supervisor.api.security] /supervisor/logs access from cebe7a76_hassio_google_drive_backup[0m

 ```
 ###### Home Assistant Core Logs:
 ```
 
During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 278, in async_update_ha_state
    await self.async_device_update()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 471, in async_device_update
    await self.hass.async_add_executor_job(self.update)  # type: ignore
  File "/usr/local/lib/python3.8/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/usr/src/homeassistant/homeassistant/components/spotify/media_player.py", line 195, in wrapper
    result = func(self, *args, **kwargs)
  File "/usr/src/homeassistant/homeassistant/components/spotify/media_player.py", line 462, in update
    current = self._spotify.current_playback()
  File "/usr/local/lib/python3.8/site-packages/spotipy/client.py", line 1635, in current_playback
    return self._get("me/player", market=market, additional_types=additional_types)
  File "/usr/local/lib/python3.8/site-packages/spotipy/client.py", line 284, in _get
    return self._internal_call("GET", url, payload, kwargs)
  File "/usr/local/lib/python3.8/site-packages/spotipy/client.py", line 239, in _internal_call
    response = self._session.request(
  File "/usr/local/lib/python3.8/site-packages/requests/sessions.py", line 530, in request
    resp = self.send(prep, **send_kwargs)
  File "/usr/local/lib/python3.8/site-packages/requests/sessions.py", line 643, in send
    r = adapter.send(request, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/requests/adapters.py", line 529, in send
    raise ReadTimeout(e, request=request)
requests.exceptions.ReadTimeout: HTTPSConnectionPool(host='api.spotify.com', port=443): Read timed out. (read timeout=5)
2020-11-02 14:31:02 WARNING (MainThread) [homeassistant.helpers.entity] Update of fan.cisticka_vzduchu is taking over 10 seconds
2020-11-02 14:41:44 ERROR (MainThread) [custom_components.hacs.factory] maykar/custom-header - Repository is archived.

 ```

**Comment by sabeechen** [comment]
That does sound like a bug.  Could you elaborate on what you mean by:

> after syncing one of old local snapshots to google drive

I'd like to try and reproduce this issue myself, but I'm unsure what you mean by that.  Were you doing anything else immediately prior to the error showing up?

Also, if you could explain what happens when you visit the folder chooser (eg why it is broken for you) I can look into that as well.

**Comment by Luc3as** [comment]
I solved the issue with folder selector, it was my privacybadge addon blocking some cookies in browser. so selecting the folder now works. I have 20 local snaphshots and I increased the google drive from 1 to 20 aswell, so now the addon tries to sync it up there. 

In the log it goes like this, 
![image](https://user-images.githubusercontent.com/6381468/97897595-fc1b0200-1d36-11eb-98af-3ba631e316ef.png)
It uploads one older snapshot, and then throws an error, when I enter the same folder again, it goes for another snapshot and upload it okay.

**Comment by sabeechen** [comment]
I've verified this is a bug.  The issue is that if you _don't_ use the "choose folder" link to authorize the addon to read the folder, Google Drive doesn't give the addon the permission it was expecting.  The workaround for right now (as you've discovered) is to select the folder through the "Choose Folder" link until I can release a fix.

Rant:
The Google Drive API is super strange.  If the addon doesn't have permission to see a folder, its still allowed to create files in the folder and query for files inside the folder so long as it has permission to see the children in the folder.  Thats why the first sync works fine (when it assumes the folder is valid) but then fails on subsequent sync's when it attempts to verify the folder is still there.  That seems like it should be a permissions error but .... I guess its the way Google intends it to work?
/Rant

**Comment by marekorok** [comment]
Hi,
just fyi, I have experienced the same problem when created folder manually in the GoogleDrive and was trying to use it just by copying the ID.
It was working only one time then error occurred, and again and again...

Using the 'choose folder' routine solves it permanently, so very likely as you said some permissions issue.

**Comment by sabeechen** [comment]
I've fixed this in the latest version, 0.103.0.  you should be able to paste any folder Id you like without having to go through the "choose folder" button.  Please let me know if you run into any more trouble with it.
